src/resolveInlineRef.ts

```diff
@@ -0,0 +1,45 @@
+import { Dictionary } from '@stoplight/types';
+import { extractSourceFromRef } from './extractSourceFromRef';
+import { pointerToPath } from './pointerToPath';
+
+function isObject(maybeObj: unknown): maybeObj is { [key in PropertyKey]: unknown } {
+  return typeof maybeObj === 'object' && maybeObj !== null;
+}
+
+function _resolveInlineRef(document: Dictionary<unknown>, ref: string, seen: Set<unknown>): unknown {
+  const source = extractSourceFromRef(ref);
+  if (source !== null) {
+    throw new ReferenceError('Cannot resolve external references');
```

Comment:
Kind of heavy handed to fail the entire process out if the document has a non inline $ref - why not ignore and continue resolving inline $refs?

Reply:
Hmm, there is no way it could continue. It doesn't resolve all $refs. It only resolves one $ref you specify.
It's a drop-in replacement for `lodash.get` calls we had in a few spots.

This is an example usage from one of the tests I included in this PR.

```ts
  test('should follow refs', () => {
    const doc = {
      a: {
        $ref: '#/b/foo',
      },
      b: {
        $ref: '#/c',
      },
      c: {
        $ref: '#/d',
      },
      d: {
        foo: 'woo!',
      },
    };

    expect(resolveInlineRef(doc, '#/a')).toEqual('woo!');
  });
```